chef/cookbooks/glance/recipes/ceph.rb

```diff
@@ -17,38 +17,47 @@
 # limitations under the License.
 #
 
-ceph_conf = node[:glance][:rbd][:store_ceph_conf]
-admin_keyring = node[:glance][:rbd][:store_admin_keyring]
-
 ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
 ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
 if ceph_servers.length > 0
   include_recipe "ceph::keyring"
+
+  ceph_conf = "/etc/ceph/ceph.conf"
+  admin_keyring = "/etc/ceph/ceph.client.admin.keyring"
+  # Ensure that the ceph config file that will be written in the glance config
+  # file is the right one
+  node.default[:glance][:rbd][:store_ceph_conf] = ceph_conf
```

Comment:
Oh, hrm. I don't know, then :-) I guess we'll see how it breaks if someone decides to trigger this nice edge case ;-)

I'll try to setup a cloud and trigger this code, to see how it goes.